### PR TITLE
Randomize donations on the fundraising index page

### PR DESCRIPTION
**Pull request by bmispelon**
The randomization is weighted by the donation amount so
that bigger donations have more chances to appear at the top.

This is basically a revert of f1e067e6ccee3a39622042a61badfbf90bde07a9.
Refs #334

### CODE REVIEW

File: fundraising/templatetags/fundraising_extras.py [hunk 1]
```diff
@@ -29,3 +30,14 @@ def donation_snippet():
         donation = None
 
     return {'donation': donation}
+
+
+@register.filter
+def shuffle_donations(donations, random=random):
+    """
+    Randomize the give list of donations but in a way that big donations
```
Comment (charettes) [inline fundraising/templatetags/fundraising_extras.py]: the given list

File: fundraising/tests.py [hunk 2]
```diff
@@ -29,9 +38,18 @@ def test_in_period_ordering(self):
...
         Donation.objects.create(donor=h3, amount='10')
-        today = date.today()
-        self.assertQuerysetEqual(
-            DjangoHero.objects.in_period(today, today + timedelta(days=1)),
-            [Decimal('15.00'), Decimal('5.00'), Decimal('10.00')],
-            attrgetter('donated_amount')
-        )
+        self.today = date.today()
+
+    def test_donation_suffling(self):
```
Comment (jezdez) [inline fundraising/tests.py]: Please don't pass in a random object but use mock instead, like here: http://stackoverflow.com/a/22457486

File: fundraising/templatetags/fundraising_extras.py [hunk 3]
```diff
@@ -29,3 +30,14 @@ def donation_snippet():
         donation = None
 
     return {'donation': donation}
+
+
+@register.filter
+def shuffle_donations(donations, random=random):
```
Comment (jezdez) [inline fundraising/templatetags/fundraising_extras.py]: I'd rather see that applied in the view than the template layer to reduce effort to port this over to Jinja2 later

File: fundraising/tests.py (hunk 2 above)
Comment (bmispelon) [inline fundraising/tests.py]: `mock` isn't currently installed though, right?

Comment (jezdez) [inline fundraising/tests.py]: nope, add it to the `requirements/tests.txt`

**Comment by bmispelon** [comment]
Switched to `mock` and moved the logic inside the view rather than the template.

**Pull request merged**